fdb-extensions/src/main/java/com/apple/foundationdb/async/rabitq/FhtKacRotator.java

```diff
@@ -0,0 +1,241 @@
+/*
+ * FhtKacRotator.java
+ *
+ * This source file is part of the FoundationDB open source project
+ *
+ * Copyright 2015-2025 Apple Inc. and the FoundationDB project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.apple.foundationdb.async.rabitq;
+
+import com.apple.foundationdb.async.hnsw.DoubleVector;
+import com.apple.foundationdb.async.hnsw.Vector;
+
+import javax.annotation.Nonnull;
+import java.util.Arrays;
+import java.util.Random;
+
+/** FhtKac-like random orthogonal rotator.
+ *  - R rounds (default 4)
+ *  - Per round: random ±1 -> FWHT on largest 2^k block (head/tail alternation) -> π/4 Givens across halves
+ *  Time per apply: O(R * (n log n)) with tiny constants; memory: O(R * n) bits for signs.
+ */
+@SuppressWarnings({"checkstyle:MethodName", "checkstyle:MemberName"})
+public final class FhtKacRotator implements LinearOperator {
```

Comment:
I think we probably need a bit more context as to what `FhtKac` means, as I was 50/50 whether a cat went over your keyboard

Reply:
I wrote a bunch of documentation.

Reply:
I think the documentation misses what "FHT" actually means, and you need to do a bit of work to associate the "K" in the name with the `K` matrix mentioned on line 49 of the Javadoc, if those are the same. It might be good to spell that out in the introductory paragraph. Otherwise, I like the explanation here!